src/plugins/workspace/server/saved_objects/workspace_saved_objects_client_wrapper.ts

```diff
@@ -128,6 +129,36 @@ export class WorkspaceSavedObjectsClientWrapper {
     return false;
   }
 
+  private async validateWorkspacesAndSavedObjectsPermissions(
+    savedObject: Pick<SavedObject, 'id' | 'type' | 'workspaces' | 'permissions'>,
+    request: OpenSearchDashboardsRequest,
+    workspacePermissionModes: WorkspacePermissionMode[],
+    objectPermissionModes: WorkspacePermissionMode[],
+    validateAllWorkspaces = true
+  ) {
+    const { id, type } = savedObject;
+    if (!savedObject.workspaces && !savedObject.permissions) {
```

Comment:
Since this is used for an edge case, can we add a comment to explain which cases will be used for this?

Reply:
Is that means add an annotation like "Ignore permission check when saved objects no workspaces and no permissions"? I think the code is very clear here. This annotation is unnecessary.

Reply:
I just mean that a object with no workspaces and permissions is not a regular case. It will be better if we can explain when a lack of workspace and permissions occurs at the same time.

Reply:
So, is that means add an annotation like "Skip permission check for Advanced settings saved objects etc."? From my opinion, this skip is not just for the `Advanced settings` object. The validate will always return `true` if saved object without workspaces and permissions. This judgement isn't complicated here. Could you give some annotation examples?

Reply:
My origin thought is that we can add a comment that `Advanced settings have no permissions and workspaces, so we need to skip it.` to elaborate it because there is nowhere to describe this particular case. Although other objects with no permissions and workspaces will also be skipped in this function, but this is mostly some case that we don't care or dirty data which we can't control, right?

Reply:
Of course, it depends on your own style.